core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalSystem.java

```diff
@@ -1031,7 +1032,9 @@ public void format() throws IOException {
       if (journalPath.exists()) {
         FileUtils.delete(journalPath.getAbsolutePath());
       }
-      journalPath.mkdirs();
+      if (!journalPath.mkdirs()) {
+        throw new AccessDeniedException(journalPath.getPath());
```

Comment:
why `AccessDeniedException`?

Reply:
There is no privilege to operate the journal folder. Do you have good advice?

Reply:
It's kind of misleading as deletion could have failed due to any reason.
I'd rather use `org.apache.commons.io.FileUtils.forceDelete` above to make sure deletion is succeeded.

Reply:
Thanks for your review. 
I compare `org.apache.commons.io.FileUtils.forceDelete` and `alluxio.util.io.FileUtils.delete`. I find they both invoke `java.nio.file.deleteIfExists`. There is no extra operation logic of `org.apache.commons.io.FileUtils.forceDelete` for a  journal path which is a file.

Reply:
I still don't understand throwing an exception when mkdir returns false.
`FileUtils.cleanDirectory()` will not delete the dir but just clean its children. That means `journalPath.mkdirs()` will return false according to javadoc below. So we'll be throwing for when we should not. 

> 
Returns: true if and only if the directory was created, along with all necessary parent directories; false otherwise

Reply:
Maybe I don't understand your discussion.
First we set alluxio.master.journal.folder. The journalPath is `${alluxio.master.journal.folder}/JobJournal`.

`if (journalPath.isDirectory()) {
      FileUtils.cleanDirectory(mConf.getPath()); // When journalPath is a  directory, it just cleans its children.
    } else {  // When journalPath is not a directory, enter this code logic
      if (journalPath.exists()) { 
        FileUtils.forceDelete(journalPath);  // When then file journalPath exists. Delete it.
      }
      if (!journalPath.mkdirs()) { // if `alluxio.master.journal.folder` is a no privileged path, return false.
        throw new AccessDeniedException(journalPath.getPath()); 
      }
    }`

Reply:
Oh. OK. I was misreading the braces.